docs/jetstream.md

```diff
@@ -0,0 +1,213 @@
+# NATS JetStream - Proposal Specification
+
+This document outlines the proposed implementation of NATS JetStream channel support.
+
+## Overview
+
+There are two components to this implementation:
+
+- Controller - `jetstream-channel-controller`
+- Dispatcher - `jetstream-channel-dispatcher`
+
+### Controller
+
+The controller reconciles `NatsJetStreamChannel` CRDs by ensuring the `Deployment` named `jetstream-ch-dispatcher`
+exists and ensures a `Service` corresponding to the channel exists with an `externalName` pointing to the
+`jetstream-ch-dispatcher`.
+
+#### Scoping
+
+`NatsJetStreamChannel` resources may be scoped to a namespace by adding the annotation
+`eventing.knative.dev/scope: namespace`, allowing the use of dedicated dispatcher deployments per namespace. This has
+two main benefits:
+
+- Channels across different namespaces cannot be starved of resources based on channels in other namespaces.
+- Channels across different namespaces can be configured with different credentials.
+
+#### Controller Dispatcher Configuration
```

Comment:
Mark this section as a future enhancement, for now the config map must be named `config-nats` and there is no way to customise it